spec.md

```diff
@@ -680,15 +680,23 @@ All registers are linked to a clock (see [@sec:registers]).
 ### Reset Types
 
 Once a circuit is powered on, it may require an explicit reset in order to put it into a known state.
-For this, we use a reset type.
-In FIRRTL, we have the option of using both synchronous or asynchronous resets.
+For this, registers take a reset signal of a reset type.  The reset type
+determines the implementation of the registers.
 
-The synchronous reset type is simply a 1-bit unsigned integer: `UInt<1>`{.firrtl}.
-The asynchronous reset type is `AsyncReset`{.firrtl}.
+There are 3 reset behaviors captured by reset type:
+* Synchronous resets (`sync`{.firrtl}) - These are sampled for activation on the rising clock edge.
+* Asynchronous resets (`async`{.firrtl}) - These become active (and inactive) immediately without regard to a clock.
+* Asynchronous activation, synchronous deactivation (`asyncsync`{.firrtl}) - These become active immediate without regard to a clock, but deactivate on the clock edge after the reset is no longer asserted.
```

Comment:
It may be useful to be very precise here. There's four possibilities:

1. sync assert, sync deassert
2. async assert, sync deassert
3. sync assert, async deassert
4. async assert, async deassert

Only (1) and (2) are useful from a digital design perspective. (4) is useful when modeling the real world (this is a button press) and (3) seems very odd, but might come up in the real world.

Currently, this PR is defining (1), (4), and (2), respectively. Should the PR skip (4) here and only talk about (2)?

tl;dr: I would suggest defining `async` as `asyncsync` and then not mentioning what is called `async` today. My reasoning here is that this is not particularly useful.

The caveat for doing this, which I expect is your motivation here, is that you need what the PR is calling `async` in order to write the actual reset synchronizer circuit (e.g., `rst_n_in` input in this circuit: https://github.com/seldridge/verilog/blob/69b1a206b1ccd8f1ca8cdb6482c59292d6700b9c/src/reset.v).

Given that, this is likely exactly what we want.

Reply:
I agree with Schuyler here as well.

Reply:
simplified.